src/test/ui/generator/issue-93161.rs

```diff
@@ -32,7 +32,7 @@ fn never() -> Never {
 }
 
 async fn includes_never(crash: bool, x: u32) -> u32 {
-    let mut result = async { x * x }.await;
+    let result = async { x * x }.await;
```

Comment:
To be clear: Wesley's assumption is that the unused `mut` lint started firing here (because the mutation of `result` falls after the call to `never()` below, and so that's why the PR author has removed the `mut`.

And Wesley's question is: Do we need to worry about this? Considering e.g. code that might have `#![deny(unused_mut)]`, and now it starts seeing that as an error.

(My personal answer is that lints are allowed to change their behavior in this manner. I'm currently double-checking/asking around for examples from Rust's past.)

Reply:
Additional discussion in https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/lints.2C.20flow-sensitivity.2C.20and.20stability

Reply:
The lint side of things seems fine to me from compatibility perspective. It is
not unusual for us to adjust how lints work, and practical impact is limited by
cargo capping the lint level for dependencies.

The fact that we would now accept more programs seems more significant, since
it would be a breaking change to revert. Though, as far as I can see, we did in
fact accept those programs after #54125 landed, but have since regressed after
switching to MIR borrowck.

Reply:
It also seems surprising that this `mut` would be unused, since `result` *is* mutated in later code -- in a regular function, we are not that smart wrt unused-mut earnings, or are we?

Reply:
I was talking about this with @scottmcm a little while back and he found this example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=26220b66188903b1dd8f3325960f75af

That version compiles even though `x` is mutated, since the mutation comes after the panic. Although, for me making it `let mut x` doesn't raise an unused mut warning. Anyway, I think this shows prior art for being clever about things being mutated only in unreachable code, so this change seems to make things more uniform.